components/net/cookie_storage.rs

```diff
@@ -37,6 +37,23 @@ impl CookieStorage {
         let domain = reg_host(cookie.cookie.domain.as_ref().unwrap_or(&"".to_string()));
         let cookies = self.cookies_map.entry(domain).or_insert(vec![]);
 
+        // https://www.ietf.org/id/draft-ietf-httpbis-cookie-alone-01.txt Step 2
+        if !cookie.cookie.secure && !source.is_secure_protocol() &&
+           cookies.iter().any(|c| {
+                let path_match = if let (&Some(ref new), &Some(ref existing)) = (&cookie.cookie.path, &c.cookie.path) {
+                    Cookie::path_match(new, existing)
+                } else {
+                    cookie.cookie.path == c.cookie.path
+                };
+
+                c.cookie.name == cookie.cookie.name &&
+                c.cookie.secure &&
+                c.cookie.domain == cookie.cookie.domain &&
```

Comment:
I was thinking that bi-directional domain-matching is equivalent to equality comparison. [RFC 2965](https://tools.ietf.org/search/rfc2965#section-1) states the following:
>      Host A's name domain-matches host B's if
>
>      *  their host name strings string-compare equal; or
>      *  A is a HDN string and has the form NB, where N is a non-empty
>         name string, B has the form .B', and B' is a HDN string.  (So,
>         x.y.com domain-matches .Y.com but not Y.com.)